pkg/generate/app/cmd/newapp.go

```diff
@@ -1097,17 +1097,20 @@ func (c *AppConfig) run(acceptors app.Acceptors) (*AppResult, error) {
 		}
 	}
 
-	err = c.checkCircularReferences(objects)
-	if err != nil {
-		if err, ok := err.(app.CircularOutputReferenceError); ok {
-			if len(c.To) == 0 {
-				// Output reference was generated, return error.
-				return nil, fmt.Errorf("%v, set a different tag with --to", err)
+	// Only check circular references for `oc new-build`.
+	if c.ExpectToBuild {
```

Comment:
@bparees FYI this is the main change, limiting the check for only when we're running `oc new-build`.
We could alternatively also check invocations of new-app, but then limited to printing a warning.